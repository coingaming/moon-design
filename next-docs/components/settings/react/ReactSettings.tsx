import { useState } from 'react';
import {
  LogoSportsbetShort,
  LogoBitcasinoShort,
  LogoMoonDesignShort,
  LogoBetaddaShort,
} from '@heathmont/moon-assets';
import useFeatureFlags from '../../../utils/useFeatureFlags';
import { useDocsTheme, DocsBrands } from '../../themes/DocsThemeProvider';
import useThemeTW from '../../themes/useThemesTW';
import ReactElixirSwitcher from '../common/ReactElixirSwitcher';
import SettingsButton from '../common/SettingsButton';
import DarkLightModeSwitcher from './DarkLightModeSwitch';
import RTLModeSwitch from './RTLModeSwitch';

type LogosKeys = keyof typeof Logos;

interface Props {
  toggleReactAndElixir: () => void;
  isReactEnabled: boolean;
}

const Logos = {
  sportsbet: <LogoSportsbetShort fontSize="1rem" />,
  bitcasino: <LogoBitcasinoShort fontSize="1rem" />,
  moonDesign: <LogoMoonDesignShort fontSize="1rem" />,
  betadda: <LogoBetaddaShort fontSize="1rem" />,
};

const ReactSettings: React.FC<Props> = ({
  toggleReactAndElixir,
  isReactEnabled,
}) => {
  const [isOpened, setIsOpened] = useState(false);
  const toggle = () => setIsOpened(!isOpened);

  const { toggleColorScheme, setBrand, themeKeys } = useDocsTheme();

  const [isRtlEnabled, setIsRtl] = useState(false);

  const { isFeatureElixirEnabled } = useFeatureFlags();

  const toggleRtl = () => {
    const htmlTag = document && document?.getElementsByTagName('html')[0];
    setIsRtl(!isRtlEnabled);
    const dirAttr = htmlTag.getAttribute('dir');

    if (!dirAttr) {
      htmlTag.setAttribute('dir', 'rtl');
    } else {
      htmlTag.removeAttribute('dir');
    }
  };

  const filteredThemeKeys = themeKeys.filter((themeKey: string) =>
    ['sportsbet', 'bitcasino', 'moonDesign', 'betadda'].includes(themeKey)
  );

  const {
    setBrand: setBrandTW,
    toggleMode: toggleModeTW,
    getMode,
  } = useThemeTW();

  const switchBrandHandler = (themeKey: LogosKeys) => {
    setBrand(themeKey as DocsBrands);
    setBrandTW(themeKey);
  };

  const switchModeHandler = () => {
    toggleColorScheme();
    toggleModeTW();
  };
  return (
    <>
      <SettingsButton onClick={toggle} isRtlEnabled={isRtlEnabled} />
      {isOpened && (
        <>
          <div
            className={`${
              isRtlEnabled ? `left-4` : `right-4`
            } fixed bottom-16 z-50`}
          >
            {filteredThemeKeys.map((themeKey: string) => {
              const Logo = Logos[themeKey as LogosKeys] || Logos.moonDesign;
              return (
                <button
                  key={themeKey}
                  type="button"
                  onClick={() => switchBrandHandler(themeKey as LogosKeys)}
                  className={`${
                    isRtlEnabled ? `mr-4` : `ml-4`
                  } p-2 rounded-full text-black bg-white hover:bg-gray-200 inline-flex items-center justify-center shadow-md`}
                >
                  {Logo}
                </button>
              );
            })}
          </div>
          <div
            className={`${
              isRtlEnabled ? `left-4` : `right-4`
            } fixed bottom-28 z-50`}
          >
            <DarkLightModeSwitcher
              toggle={switchModeHandler}
              isEnabled={getMode() === 'dark'}
              isRtlEnabled={isRtlEnabled}
            />
          </div>
          <div
            className={`${
              isRtlEnabled ? `left-4` : `right-4`
            } fixed bottom-40 z-50`}
          >
            <RTLModeSwitch toggle={toggleRtl} isEnabled={isRtlEnabled} />
          </div>
          <div
            className={`${
              isRtlEnabled ? `left-4` : `right-4`
            } fixed bottom-52 z-50`}
          >
            {isFeatureElixirEnabled ? (
              <ReactElixirSwitcher
                toggle={toggleReactAndElixir}
                isReactEnabled={isReactEnabled}
              />
            ) : null}
          </div>
        </>
      )}
    </>
  );
};

export default ReactSettings;
