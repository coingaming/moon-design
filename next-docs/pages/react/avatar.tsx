import React from 'react';

import { ThemeProvider, sportsbetDark } from '@heathmont/moon-themes';

export default function PageAvatar() {
  return (
    <>
      <ThemeProvider theme={sportsbetDark}>
        <h1>Avatar</h1>
      </ThemeProvider>
    </>
  );
}
