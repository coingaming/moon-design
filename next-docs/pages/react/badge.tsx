import React from 'react';
import Head from 'next/head';

import { Badge } from '@heathmont/moon-core';
import { ThemeProvider, sportsbetDark } from '@heathmont/moon-themes';

export default function PageBadge() {
  return (
    <>
      <h1>Badge</h1>
      <p>Small count and labeling component.</p>
      <Badge>Active</Badge>
      <Badge color="goten.100" backgroundColor="piccolo.100">
        Active
      </Badge>
    </>
  );
}
