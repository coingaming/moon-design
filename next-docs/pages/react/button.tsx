import React from 'react';
import Head from 'next/head';

import { Button } from '@heathmont/moon-components';
import PreviewSwitch from '../../components/codePreview/Switch';
import Preview from '../../components/codePreview/Preview';

export default function PageBadge() {
  return (
    <>
      <section className="mt-6">
        <h1 className="text-5xl font-semibold">Button</h1>
        <p className="text-lg mt-4">
          Buttons are calls-to-action used to prompt users. They encourage users
          to interact with us in multiple ways throughout our galaxy, based on
          what the label of the button indicates.
        </p>

        <p className="text-lg mt-4">
          Buttons are clickable elements with label text that describe the
          action that will happen when the users interact with it.
        </p>
      </section>
      <section className="mt-6">
        <Preview
          title="Primary"
          preview={<Button variant="primary">Active</Button>}
          code={`import { Button } from '@heathmont/moon-components';

<Button variant="primary">Active</Button>`
}
        />
      </section>
    </>
  );
}
