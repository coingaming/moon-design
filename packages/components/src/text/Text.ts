// DEPRECATED! Use one from core/src/text
import styled from 'styled-components';
import { themed } from '@heathmont/moon-utils';
import { ColorNames } from '@heathmont/moon-themes';

import getFontSize from '../private/text/getFontSize';

export type Size = 10 | 12 | 14 | 16 | 18 | 20 | 24 | 32 | 48 | 56 | 64 | 72;

type Props = {
  size?: Size;
  color?: ColorNames;
  isBold?: boolean;
};

const Text = styled.p<Props>(
  ({ color, theme }) => ({
    color: themed('color', color)(theme),
  }),
  ({ size = 16 }) => getFontSize(size),
  ({ isBold, theme: { fontWeight } }) =>
    isBold && {
      fontWeight: fontWeight.semibold,
    }
);

Text.defaultProps = {
  color: 'bulma.100',
  size: 16,
};

export default Text;
