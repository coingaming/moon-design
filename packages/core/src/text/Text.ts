import { ColorNames } from '@heathmont/moon-themes';
import { themed } from '@heathmont/moon-utils';
import styled from 'styled-components';

import getFontSize from '../private/text/getFontSize';

export type Size = 10 | 12 | 14 | 16 | 18 | 20 | 24 | 32 | 48 | 56 | 64 | 72;

export interface TextProps {
  size?: Size;
  color?: ColorNames;
  isBold?: boolean;
  textAlign?: TextAlign;
}

const Text = styled.p<TextProps>(
  ({ color, theme }) => ({
    color: themed('color', color)(theme),
  }),
  ({ size = 16 }) => getFontSize(size),
  ({ textAlign }) =>
    textAlign && {
      textAlign,
    },
  ({ isBold, theme: { newTokens } }) =>
    isBold && {
      fontWeight: newTokens.font.weight.semibold,
    }
);

Text.defaultProps = {
  color: 'bulma.100',
  size: 16,
};

export default Text;
