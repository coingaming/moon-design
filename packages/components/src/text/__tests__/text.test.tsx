import * as React from 'react';
import { create } from 'react-test-renderer';
import 'jest-styled-components';
import { sportsbetDark, ThemeProvider } from '@heathmont/moon-themes';

import Text from '../Text';

const renderWithTheme = (component: JSX.Element) => (
  <ThemeProvider theme={sportsbetDark}>{component}</ThemeProvider>
);

describe('Text', () => {
  describe('Elements', () => {
    test('renders as <p> if no "as" prop provided', () => {
      const text = create(renderWithTheme(<Text>Text</Text>));
      expect(text).toMatchSnapshot();
    });

    test('renders as an element if "as" prop provided', () => {
      const text = create(renderWithTheme(<Text as="span">Text</Text>));
      expect(text).toMatchSnapshot();
    });
  });

  describe('Sizes', () => {
    test('renders default font size', () => {
      const text = create(renderWithTheme(<Text>Text</Text>));
      expect(text).toMatchSnapshot();
    });

    test('10rem size renders properly', () => {
      const text = create(renderWithTheme(<Text size={10}>Text</Text>));
      expect(text).toMatchSnapshot();
    });

    test('12rem size renders properly', () => {
      const text = create(renderWithTheme(<Text size={12}>Text</Text>));
      expect(text).toMatchSnapshot();
    });

    test('14rem size renders properly', () => {
      const text = create(renderWithTheme(<Text size={14}>Text</Text>));
      expect(text).toMatchSnapshot();
    });

    test('16rem size renders properly', () => {
      const text = create(renderWithTheme(<Text size={16}>Text</Text>));
      expect(text).toMatchSnapshot();
    });

    test('18rem size renders properly', () => {
      const text = create(renderWithTheme(<Text size={18}>Text</Text>));
      expect(text).toMatchSnapshot();
    });

    test('20rem size renders properly', () => {
      const text = create(renderWithTheme(<Text size={20}>Text</Text>));
      expect(text).toMatchSnapshot();
    });

    test('24rem size renders properly', () => {
      const text = create(renderWithTheme(<Text size={24}>Text</Text>));
      expect(text).toMatchSnapshot();
    });

    test('32rem size renders properly', () => {
      const text = create(renderWithTheme(<Text size={32}>Text</Text>));
      expect(text).toMatchSnapshot();
    });

    test('48rem size renders properly', () => {
      const text = create(renderWithTheme(<Text size={48}>Text</Text>));
      expect(text).toMatchSnapshot();
    });

    test('56rem size renders properly', () => {
      const text = create(renderWithTheme(<Text size={56}>Text</Text>));
      expect(text).toMatchSnapshot();
    });

    test('64rem size renders properly', () => {
      const text = create(renderWithTheme(<Text size={64}>Text</Text>));
      expect(text).toMatchSnapshot();
    });

    test('72rem size renders properly', () => {
      const text = create(renderWithTheme(<Text size={72}>Text</Text>));
      expect(text).toMatchSnapshot();
    });

    describe('Colors', () => {
      test('renders default color', () => {
        const text = create(renderWithTheme(<Text>Text</Text>));
        expect(text).toMatchSnapshot();
      });

      test('renders with a color', () => {
        const text = create(
          renderWithTheme(<Text color="piccolo.100">Text</Text>)
        );
        expect(text).toMatchSnapshot();
      });
    });

    describe('Boldness', () => {
      test('renders default boldness', () => {
        const text = create(renderWithTheme(<Text>Text</Text>));
        expect(text).toMatchSnapshot();
      });

      test('renders with bold font weight', () => {
        const text = create(renderWithTheme(<Text isBold>Text</Text>));
        expect(text).toMatchSnapshot();
      });
    });
  });
});
