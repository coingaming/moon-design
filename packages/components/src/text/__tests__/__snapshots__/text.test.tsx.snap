// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Elements renders as <p> if no "as" prop provided 1`] = `
.c0 {
  color: #FFFFFF;
  font-size: 1rem;
  line-height: 1.5rem;
}

<p
  className="c0"
  color="bulma.100"
  size={16}
>
  Text
</p>
`;

exports[`Text Elements renders as an element if "as" prop provided 1`] = `
.c0 {
  color: #FFFFFF;
  font-size: 1rem;
  line-height: 1.5rem;
}

<span
  className="c0"
  color="bulma.100"
  size={16}
>
  Text
</span>
`;

exports[`Text Sizes 10rem size renders properly 1`] = `
.c0 {
  color: #FFFFFF;
  font-size: 0.625rem;
  line-height: 1rem;
}

<p
  className="c0"
  color="bulma.100"
  size={10}
>
  Text
</p>
`;

exports[`Text Sizes 12rem size renders properly 1`] = `
.c0 {
  color: #FFFFFF;
  font-size: 0.75rem;
  line-height: 1rem;
}

<p
  className="c0"
  color="bulma.100"
  size={12}
>
  Text
</p>
`;

exports[`Text Sizes 14rem size renders properly 1`] = `
.c0 {
  color: #FFFFFF;
  font-size: 0.875rem;
  line-height: 1.25rem;
}

<p
  className="c0"
  color="bulma.100"
  size={14}
>
  Text
</p>
`;

exports[`Text Sizes 16rem size renders properly 1`] = `
.c0 {
  color: #FFFFFF;
  font-size: 1rem;
  line-height: 1.5rem;
}

<p
  className="c0"
  color="bulma.100"
  size={16}
>
  Text
</p>
`;

exports[`Text Sizes 18rem size renders properly 1`] = `
.c0 {
  color: #FFFFFF;
  font-size: 1.125rem;
  line-height: 1.75rem;
}

<p
  className="c0"
  color="bulma.100"
  size={18}
>
  Text
</p>
`;

exports[`Text Sizes 20rem size renders properly 1`] = `
.c0 {
  color: #FFFFFF;
  font-size: 1.25rem;
  line-height: 1.75rem;
}

<p
  className="c0"
  color="bulma.100"
  size={20}
>
  Text
</p>
`;

exports[`Text Sizes 24rem size renders properly 1`] = `
.c0 {
  color: #FFFFFF;
  font-size: 1.5rem;
  line-height: 2rem;
}

<p
  className="c0"
  color="bulma.100"
  size={24}
>
  Text
</p>
`;

exports[`Text Sizes 32rem size renders properly 1`] = `
.c0 {
  color: #FFFFFF;
  font-size: 2rem;
  line-height: 2.5rem;
}

<p
  className="c0"
  color="bulma.100"
  size={32}
>
  Text
</p>
`;

exports[`Text Sizes 48rem size renders properly 1`] = `
.c0 {
  color: #FFFFFF;
  font-size: 3rem;
  line-height: 3.5rem;
}

<p
  className="c0"
  color="bulma.100"
  size={48}
>
  Text
</p>
`;

exports[`Text Sizes 56rem size renders properly 1`] = `
.c0 {
  color: #FFFFFF;
  font-size: 3.5rem;
  line-height: 4rem;
}

<p
  className="c0"
  color="bulma.100"
  size={56}
>
  Text
</p>
`;

exports[`Text Sizes 64rem size renders properly 1`] = `
.c0 {
  color: #FFFFFF;
  font-size: 4rem;
  line-height: 4.5rem;
}

<p
  className="c0"
  color="bulma.100"
  size={64}
>
  Text
</p>
`;

exports[`Text Sizes 72rem size renders properly 1`] = `
.c0 {
  color: #FFFFFF;
  font-size: 4.5rem;
  line-height: 4.75rem;
}

<p
  className="c0"
  color="bulma.100"
  size={72}
>
  Text
</p>
`;

exports[`Text Sizes Boldness renders default boldness 1`] = `
.c0 {
  color: #FFFFFF;
  font-size: 1rem;
  line-height: 1.5rem;
}

<p
  className="c0"
  color="bulma.100"
  size={16}
>
  Text
</p>
`;

exports[`Text Sizes Boldness renders with bold font weight 1`] = `
.c0 {
  color: #FFFFFF;
  font-size: 1rem;
  line-height: 1.5rem;
  font-weight: 600;
}

<p
  className="c0"
  color="bulma.100"
  size={16}
>
  Text
</p>
`;

exports[`Text Sizes Colors renders default color 1`] = `
.c0 {
  color: #FFFFFF;
  font-size: 1rem;
  line-height: 1.5rem;
}

<p
  className="c0"
  color="bulma.100"
  size={16}
>
  Text
</p>
`;

exports[`Text Sizes Colors renders with a color 1`] = `
.c0 {
  color: #0CD463;
  font-size: 1rem;
  line-height: 1.5rem;
}

<p
  className="c0"
  color="piccolo.100"
  size={16}
>
  Text
</p>
`;

exports[`Text Sizes renders default font size 1`] = `
.c0 {
  color: #FFFFFF;
  font-size: 1rem;
  line-height: 1.5rem;
}

<p
  className="c0"
  color="bulma.100"
  size={16}
>
  Text
</p>
`;
